lib/Conversion/FIRRTLToHW/LowerToHW.cpp

```diff
@@ -3423,6 +3423,20 @@ Value FIRRTLLowering::createArrayIndexing(Value array, Value index) {
   auto size = hw::type_cast<hw::ArrayType>(array.getType()).getSize();
   auto valWire = builder.create<sv::WireOp>(
       hw::type_cast<hw::ArrayType>(array.getType()).getElementType());
+
+  // Extend to power of 2
+  if (!llvm::isPowerOf2_64(size)) {
+    auto extElem =
+        builder.create<hw::ConstantOp>(APInt(llvm::Log2_64_Ceil(size), 0));
```

Comment:
This should be able to consult the constant pool:

```suggestion
    auto extElem = getOrCreateIntConstant(APInt(llvm::Log2_64_Ceil(size), 0));
```